File: oom.go
package main

import (
	"fmt"
	"os"
	"time"
)

var identifier []int
var c int = 0

func run() {
	identifier = append(identifier, 1)
	// fmt.Println("--")
	time.Sleep(10 * time.Millisecond)
	c++

	run()
}

func main() {
	done := make(chan bool)

	fmt.Printf("pid: %d\n", os.Getpid())
	for i := 0; i < 10000; i++ {
		go run()
	}

	<-done // Block forever
}

File: oom.js
var arr = [];

const sleep = (ms) => {
  return new Promise((resolve) => {
    setTimeout(resolve, ms);
  });
};

const run = () => {
  return sleep(0).then(() => {
    arr.push(10);
    return run();
  });
};

console.log(process.pid);

for (let index = 0; index < 10000; index++) {
  run();
}
